PythonPonies/bangazon-api
# Zgl authorization

# Description
Add description about what this pull request covers.
- this request covers limiting user data displayed depending on authentication
## Number of Fixes
Type of number of fixes included in this PR.
- fixed serialization, permission classes and viewset
## Related Ticket(s)
List related ticket(s) that apply to the work done in this pull request
#73 #72 #71 #64 
## Problem to Solve
Describe the problem(s) this pull request solves.
- solves the issue of displaying sensitive data on our api
## Proposed Changes
Describe the proposed changes.
- now if not logged as admin, can only see user names and id
## Expected Behavior
Describe the expected behavior.
- should still be able to navigate from the root api, but only see user first name, last name and id under users unless admin
## Steps to Test Solution

1. git fetch
2. makemigrations
3. migrate
4. runserver
5. go to root, check out users
6. sign in as admin, check out users

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[X ] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
